lib/semantics/check-coarray.cc

```diff
@@ -56,7 +57,7 @@ void CoarrayChecker::Leave(const parser::FormTeamStmt &x) {
   AnalyzeExpr(context_, std::get<parser::ScalarIntExpr>(x.t));
   const auto &teamVar{std::get<parser::TeamVariable>(x.t)};
   AnalyzeExpr(context_, teamVar);
-  const parser::Name *name{GetSimpleName(teamVar.thing)};
+  const parser::Name *name{parser::Unwrap<parser::Name>(teamVar)};
```

Comment:
It seems like GetLastName is closer to what GetSimpleName did. How should we decide between using GetLastName and Unwrap<Name>?

Reply:
`Unwrap` should be doing what `GetSimpleName` was meant to do: if a parse tree contains a node of interest, like a `Name`, possibly wrapped up in constraint checks, `Expr`, &c., `Unwrap` will find it and return it to you, otherwise null.  If you try to unwrap a Name from `X%Y`, it fails.

`GetLastName` finds the name that determines the type of an expression or variable.  It returns a reference since it can't fail.  On `X%Y`, it return `Y`.